src/Core/Grid/Query/Filter/DoctrineFilterApplicator.php

```diff
@@ -80,6 +80,37 @@ public function apply(QueryBuilder $qb, SqlFilters $filters, array $filterValues
                         $qb->setParameter($name, sprintf('%s %s', $value['to'], '23:59:59'));
                     }
 
+                    break;
+                case SqlFilters::MIN_MAX:
+                    $minFieldExists = isset($value['min_field']);
+                    $maxFieldExists = isset($value['max_field']);
+
+                    if (!$minFieldExists && !$maxFieldExists) {
+                        break;
+                    }
+
+                    $bothFieldsExist = $minFieldExists && $maxFieldExists;
+                    $minFieldExistsOnly = $minFieldExists && !$maxFieldExists;
+                    $maxFieldExistsOnly = $maxFieldExists && !$minFieldExists;
+                    $bothFieldsAreEqual = $bothFieldsExist && $value['min_field'] === $value['max_field'];
+
+                    $minFieldSqlCondition = "$sqlField >= :{$filterName}_min";
+                    $maxFieldSqlCondition = "$sqlField <= :{$filterName}_max";
+
+                    if ($bothFieldsExist && !$bothFieldsAreEqual) {
```

Comment:
I like to use `switch` statements for this kind of logic, but it's just personal preferences 😄  this is OK too

Reply:
While I like switch statements more too, is it really possible in this case? 

Each ifelse uses different variables. So we can't pick one variable and check which value it has.

Reply:
2 solutions: 1 good and 1 nasty

## Good solution

Use an integer with constants to store the usecase. Something like
```
$myUsecase = $this->computeMyUsecase($bothFieldsExist, minFieldExistsOnly, maxFieldExistsOnly, bothFieldsAreEqual);
```
and then you have a switch
```
switch($myUsecase)
    case self::BOTH_FIELS_EXIST_BUT_NOT_EQUAL: ...
```

## Nasty solution

I mention it for fun 😛  but don't implement it 😄 
```
switch (true)
    case ($bothFieldsExist && !$bothFieldsAreEqual): ...
        break;
        
    case ($minFieldExistsOnlyl): ...
        break;
        
       case (maxFieldExistsOnly): ...
        break;
```

Reply:
I quite like a solution, I used this opportunity to move rest of the logic to determine the min_max case to a separate function and it makes the code cleaner. 

Thanks!